projects/gateway2/setup/ggv2setup_test.go

```diff
@@ -240,14 +241,41 @@ func TestScenarios(t *testing.T) {
 				// that we get test pollution.
 				// once we change it to only include the ones in the proxy, we can re-enable this
 				//				t.Parallel()
-				testScenario(t, ctx, client, xdsPort, fullpath)
+				testScenario(t, ctx, setupOpts.KrtDebugger, snapCache, client, xdsPort, fullpath)
 
 			})
 		}
 	}
 }
 
-func testScenario(t *testing.T, ctx context.Context, client istiokube.CLIClient, xdsPort int, f string) {
+func dumpCache(snapCache cache.SnapshotCache) iosnapshot.SnapshotResponseData {
```

Comment:
I'm ok with this for now as it's an incremental improvement, but it feels like we're building functionality into our tests that could be exposed elsewhere. This is identical to https://github.com/solo-io/gloo/blob/main/projects/gloo/pkg/servers/iosnapshot/history.go#L184, so perhaps we need to consider creating a way to use that?